File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/model/TraitRemoteDao.java
package me.nathanp.magiccreatures.model;

import com.google.firebase.firestore.FirebaseFirestore;

import java.util.List;

import androidx.lifecycle.LiveData;

class TraitRemoteDao {

    private LiveFirestoreCollection<Trait> liveTraitCollection;

    public TraitRemoteDao() {
        liveTraitCollection = new LiveFirestoreCollection<>(FirebaseFirestore.getInstance().collection(Paths.getTraitCollectionPath()), Trait.class);
    }

    public LiveData<List<Trait>> getAllTraits() {
        return liveTraitCollection;
    }
}

File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/model/TraitRemoteDao.kt
package me.nathanp.magiccreatures.model

import androidx.lifecycle.LiveData
import com.google.firebase.firestore.FirebaseFirestore

internal class TraitRemoteDao {
    private val liveTraitCollection: LiveFirestoreCollection<Trait> = LiveFirestoreCollection(FirebaseFirestore.getInstance().collection(Paths.getTraitCollectionPath()), Trait::class.java)
    val allTraits: LiveData<List<Trait>>
        get() = liveTraitCollection
}